src/windows/main_window.py

```diff
@@ -1245,8 +1261,12 @@ def actionPreviousMarker_trigger(self, event):
             # Get selected object
             selected_clip = Clip.get(id=clip_id)
             if selected_clip:
-                all_marker_positions.append(selected_clip.data["position"])
-                all_marker_positions.append(selected_clip.data["position"] + (selected_clip.data["end"] - selected_clip.data["start"]))
+                clip_start_time=selected_clip.data["position"]
+                clip_stop_time=selected_clip.data["position"] + (selected_clip.data["end"] - selected_clip.data["start"])
+                all_marker_positions.append(clip_start_time)
+                all_marker_positions.append(clip_stop_time)
+                # add all keyframes
+                all_marker_positions.extend(getKeyframePositions(selected_clip.data, clip_start_time, clip_stop_time, fps_float))
```

Comment:
Can't you move this logic into `getKeyframePositions()` as well? It will be the same whether you've got a transition or a clip selected, either way you're adding two positions to the list:
```python3
positions.append(data["position"])
positions.append(data["position"] + (data["end"] - data["start"]))
```
If you do that _inside_ the function, then
1. You don't have to duplicate that code in the `if selected_clip:` and `if selected_tran:` sections
2. You don't need to pass `start_time` and `stop_time` as arguments to `getKeyframePositions()`. You already have everything you need in the `data` dictionary to just compute them inside the function.

Then you should only need a single call per clip/transition:
```python
all_marker_positions.extend(getKeyframePositions(selected_clip.data, fps_float))
# or
all_marker_positions.extend(getKeyframePositions(selected_tran.data, fps_float))
```

Reply:
(And yes, I missed that myself when I was whipping up my function-placement example earlier, so I'm partly at fault here.)

Reply:
Also, I supppose it would make `getKeyframePositions()` an inaccurate name for the function. Maybe something like `getSeekablePositions()` is better.